beaglebone/imu/src/__main__.py

```diff
@@ -1,26 +1,36 @@
+import Adafruit_BBIO.UART as UART
 import serial
 import asyncio
 import math
+import time
 from rover_common.aiohelper import run_coroutines
 from rover_common import aiolcm
 from rover_msgs import IMUData
+import bitstring
+
+# Picocom - picocom -b 115200 /dev/ttyS4
+# LCM_DEFAULT_URL="udpm://239.255.76.67:76?ttl=255" ./jarvis exec lcm_tools_echo IMUData "/imu_data"
 
 
 class IMU_Manager():
+    # data for clibration
+    calibration_matrix = [[0]*3]*3
+    mag_offsets = [0]*3
 
     def __init__(self):
+        UART.setup("UART4")
+
         # Mapping NMEA messages to their handlers
         self.NMEA_TAGS_MAPPER = {
             "PCHRS": self.pchrs_handler,
-            "PCHRA": self.pchra_handler
+            "PCHRA": self.pchra_handler,
+            "PCHRH": self.pchrh_handler
         }
         self.sleep = .01
-        self.max_error_count = 20
 
     def __enter__(self):
-
         self.ser = serial.Serial(
-            port='/dev/ttyTHS0',
+            port='/dev/ttyS4',
```

Comment:
beaglebone flag we run this on the jetson during integration so that's the ttyTHS0 port